### PR TITLE
Add filtering to veBal table

### PR DESCRIPTION
**Pull request by alter-eggo**
# Description
This pr adds filtering to veBal table by token, network and expiring of gauges. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [x] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

**Comment by garethfuller** [comment]
Can we get the modules on the right to align with the bottom please?
Screenshot 2023-01-13 at 16 31 47

File: src/components/contextual/pages/vebal/LMVoting/GaugesFilters.vue [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            {{ $t('gaugeFilter.moreFilters') }}
+          </div>
+        </div>
+      </div>
+    </template>
+    <template #default="{ close }">
+      <div
+        class="flex overflow-hidden flex-col py-4 px-3 w-64 rounded-lg"
+        @click="close"
+      >
```
Comment (garethfuller) [inline src/components/contextual/pages/vebal/LMVoting/GaugesFilters.vue:56]: Please use BalBtn for this. The style might not be exactly the same as Pon's designs but we have to stay consistent with buttons.

File: src/components/inputs/TokenSearchTextInput.vue [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  >
+    <BalIcon name="search" size="lg" class="mr-4" />
+    <div class="flex flex-wrap gap-2 items-center">
+      <input
+        class="dark:bg-gray-800"
+        :value="modelValue"
+        type="text"
+        :placeholder="placeholder"
+        @input="handleInput"
+      />
```
Comment (garethfuller) [inline src/components/inputs/TokenSearchTextInput.vue:29]: Please use BalTextInput for this, again we need to keep design consistent, that is why these Bal components exist.

**Review by garethfuller — CHANGES_REQUESTED** [review]

File: src/components/contextual/pages/vebal/LMVoting/GaugesFilters.vue (hunk 1 above)
Comment (garethfuller) [reply src/components/contextual/pages/vebal/LMVoting/GaugesFilters.vue:56]: If @pkattera want's to change the style of buttons we have to apply that to BalBtn and therefore everywhere.

**Comment by alter-eggo** [comment]
@garethfuller 
Screenshot 2023-01-16 at 12 59 30

it's already aligned to bottom. or you mean it's better to put it in one line with filters?

**Comment by garethfuller** [comment]
> it's already aligned to bottom. or you mean it's better to put it in one line with filters?

Yes please align it with the filters, unless @pkattera thinks otherwise. But aligning with the bottom of the title looks weird to me.

**Comment by alter-eggo** [comment]
@garethfuller I've done the fixes.
I used `BalBtn` and `BalTextInput` components, and deleted unnecessary `TokenSearchTextInput` component. 
info modules are now align with filters, I think looks better now, good catch! thanks for the review

File: src/components/_global/BalCheckbox/BalCheckbox.vue [hunk 3]
```diff
@@ -1,13 +1,13 @@
 <template>
   <div :class="['bal-checkbox', wrapperClasses]">
     <div class="flex">
-      <div class="flex items-start">
+      <div class="flex items-center">
```
Comment (garethfuller) [inline src/components/_global/BalCheckbox/BalCheckbox.vue:4]: This has implications in other areas when the checkbox label has a large amount of text. For example in high price impact joins, see screenshot:
Screenshot 2023-01-16 at 12 07 01

Please modify BalCheckbox to take a prop like "align" with a default value of center but that allows for the checkbox to align to the top as well.

File: src/components/contextual/pages/vebal/LMVoting/LMVoting.vue [hunk 4]
```diff
@@ -72,6 +83,70 @@ const hasExpiredLock = computed(
...
+});
+
+const debouncedFilterText = computed({
+  get() {
+    return tokenFilter.value;
+  },
+  set: debounce(newValue => {
+    tokenFilter.value = newValue;
+  }, 500),
+});
```
Comment (garethfuller) [inline src/components/contextual/pages/vebal/LMVoting/LMVoting.vue:108]: Not a fan of get/set computed properties, it's kind of an anti-pattern, isn't it? Is there an alternative pattern that achieves the same result without computed properties, if so what does it look like?

**Review by garethfuller — CHANGES_REQUESTED** [review]
Looks great, a few minor things.

File: src/components/_global/BalCheckbox/BalCheckbox.vue (hunk 3 above)
Comment (garethfuller) [reply src/components/_global/BalCheckbox/BalCheckbox.vue:4]: If the default is center you will have to find all the price impact use cases and add the prop to align with top.

File: src/components/contextual/pages/vebal/LMVoting/LMVoting.vue [hunk 5]
```diff
@@ -7,18 +7,29 @@ import useVotingEscrowLocks from '@/composables/useVotingEscrowLocks';
...
  */
+const tokenFilter = ref('');
+const hideExpiredGauges = ref(true);
+const activeNetworkFilters = ref<Network[]>([]);
 const activeVotingGauge = ref<VotingGaugeWithVotes | null>(null);
+const networks = {
+  [Network.MAINNET]: 'Ethereum',
+  [Network.POLYGON]: 'Polygon',
+  [Network.ARBITRUM]: 'Arbitrum',
+};
```
[the remark this answers is not in the document]
Comment (garethfuller) [reply src/components/contextual/pages/vebal/LMVoting/LMVoting.vue:32]: It also needs to include Optimism.

**Comment by pkattera** [comment]
@alter-eggo A few tweaks:
Screenshot 2023-01-16 at 12 22 20 PM

File: src/components/contextual/pages/vebal/LMVoting/LMVoting.vue (hunk 4 above)
Comment (alter-eggo) [reply src/components/contextual/pages/vebal/LMVoting/LMVoting.vue:108]: hmm, it seems that we can use `customRef` https://vuejs.org/api/reactivity-advanced.html#customref
I will try

File: src/components/_global/BalCheckbox/BalCheckbox.vue (hunk 3 above)
Comment (alter-eggo) [reply src/components/_global/BalCheckbox/BalCheckbox.vue]: fixed

File: src/components/contextual/pages/vebal/LMVoting/LMVoting.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (alter-eggo) [reply src/components/contextual/pages/vebal/LMVoting/LMVoting.vue]: moved to useNetwork

File: src/components/contextual/pages/vebal/LMVoting/LMVoting.vue (hunk 4 above)
Comment (alter-eggo) [reply src/components/contextual/pages/vebal/LMVoting/LMVoting.vue]: yeah, I used `customRef` and it works in the same way

**Comment by alter-eggo** [comment]
@garethfuller @pkattera fixed all the comments, except one by Pon:
if some filter is active I just change the color of `More filters` text and icon.
the problem is that the filter applying is debounced, so number changes only after some time (500ms in our case). it’s not easy to fix and will lead to a lot of overhead logic

File: src/components/_global/BalCheckbox/BalCheckbox.vue [hunk 6]
```diff
@@ -89,7 +90,7 @@ export default defineComponent({
         case 'lg':
           return 'w-6 h-6';
         default:
-          return 'w-5 h-5 mt-1';
```
Comment (garethfuller) [inline src/components/_global/BalCheckbox/BalCheckbox.vue:92]: This `mt-1` and the `relative top-2` above in the diff were probably required to make the items-start look right. Please conditionally add these in if the alignment is to the top.

File: src/composables/useNetwork.ts [hunk 7]
```diff
@@ -27,6 +27,13 @@ const NETWORK_ID =
 if (windowAvailable) localStorage.setItem('networkId', NETWORK_ID.toString());
 export const networkSlug = config[NETWORK_ID].slug;
 export const networkConfig = config[NETWORK_ID];
+export const networksMap = {
```
Comment (garethfuller) [inline src/composables/useNetwork.ts:30]: Since lots of things could be a 'networksMap' let's rename it 'networkLabelMap' to be more specific.

**Review by garethfuller — CHANGES_REQUESTED** [review]
> so number changes only after some time (500ms in our case)

Even if the number takes a while to update I think it would be worth adding. Is it really difficult just to track active filters even if it's not technically accurate for some time will the debounce functions happen in the background?

**Review by garethfuller — APPROVED** [review]

**Pull request merged**